frontend/src/components/buttons/ButtonDelete.vue

```diff
@@ -1,15 +1,18 @@
 <template>
   <v-btn
     :text="text"
-    class="px-3 px-sm-4"
+    class="ec-btn-delete px-3 px-sm-4"
     min-width="0"
     :color="color"
     :icon="iconOnly"
     v-bind="$attrs"
     v-on="$listeners"
   >
     <v-icon v-if="iconOnly">{{ icon }}</v-icon>
-    <v-icon v-else :left="$vuetify.breakpoint.smAndUp" size="150%">{{ icon }}</v-icon>
+    <v-icon v-else-if="!text" :left="$vuetify.breakpoint.smAndUp">{{ icon }}</v-icon>
+    <v-icon v-if="!iconOnly && text" :left="$vuetify.breakpoint.smAndUp">{{
```

Comment:
If I understand correctly, you could use `v-else` here, I think that might be easier to understand?
```suggestion
    <v-icon v-else :left="$vuetify.breakpoint.smAndUp">{{
```